reactive/context_test.ts

```diff
@@ -183,3 +215,486 @@ test("all")("Scope.target works as expected when run within a `with` context", (
+  expect(c.target.nested.func()).toBe(2)
+
+  // Delete the entire nested object in context `b`.
+  delete b.target.nested
+
+  // Validate that the nested property `b` remains in context `a` and `c`.
+  expect(a.target.nested.b).toBe(0)
+  expect(b.target.nested).toBeUndefined()
+  // @ts-ignore Typescript doesn't like deep nested properties for some reason
+  expect(c.target.nested.b).toBe(2)
+
+  // Validate that the nested objects are correctly removed from context `b`.
+  expect(Object.keys(a.target).sort()).toEqual(["d", "nested"])
+  expect(Object.keys(b.target).sort()).toEqual(["d"])
+  expect(Object.keys(c.target).sort()).toEqual(["nested"])
+
+  // Ensure the nested properties are correct after the operations.
+  expect(Object.keys(a.target.nested).sort()).toEqual(["b", "func"])
+  expect(Object.keys(c.target.nested).sort()).toEqual(["b", "func"])
+
+  // Finally, delete the nested object in context `c`.
+  delete c.target.nested
+
+  // Ensure that only `a` retains the nested property.
+  expect(a.target.nested.b).toBe(0)
+  expect(b.target.nested).toBeUndefined()
+  expect(c.target.nested).toBeUndefined()
+
+  // Validate the presence of `nested` in context `a` but not in `b` or `c`.
+  expect("nested" in a.target).toBe(true)
+  expect("nested" in b.target).toBe(false)
+  expect("nested" in c.target).toBe(false)
+
+  // Ensure that `nested` still exists in context `a`.
+  expect(a.target).toHaveProperty("nested")
+  expect(b.target).not.toHaveProperty("nested")
+  expect(c.target).not.toHaveProperty("nested")
+
+  // Validate the final structure after all deletions.
+  expect(Object.keys(a.target).sort()).toEqual(["d", "nested"])
+  expect(Object.keys(b.target).sort()).toEqual(["d"])
+  expect(Object.keys(c.target).sort()).toEqual([])
+
+  // Ensure that the nested object structure remains intact in context `a`.
+  expect(Object.keys(a.target.nested).sort()).toEqual(["b", "func"])
+})
+
+test("all")("Scope.target works with Set, Map, Date, and ArrayBuffer", () => {
+  const { observable, context } = observe({
+    setOfUrls: new Set(["https://example.com"]),
+    date: new Date(),
+    buffer: new ArrayBuffer(16),
+    myMap: new Map([["key", "value"]]),
+  })
+
+  const childContext = context.with({
+    date: new Date("2025-01-01T00:00:00Z"),
+    buffer: new ArrayBuffer(32),
+  })
+
+  expect(observable.setOfUrls.has("https://example.com")).toBe(true)
+  expect(childContext.target.setOfUrls.has("https://example.com")).toBe(true)
+
+  expect(observable.buffer.byteLength).toBe(16)
+  expect(childContext.target.buffer.byteLength).toBe(32)
+
+  const date = childContext.target.date
+  expect(date.toISOString()).toBe("2025-01-01T00:00:00.000Z")
+
+  childContext.target.myMap.set("newKey", "newValue")
+  expect(observable.myMap.has("newKey")).toBe(true)
+  expect(observable.myMap.get("newKey")).toBe("newValue")
+})
+
+test("all")("Scope.with() creates isolated and shared Date and ArrayBuffer", () => {
```

Comment:
I'm not sure what this test brings compared to previous test.

The following is already tested in previous test case:
```ts
  expect(observable.buffer.byteLength).toBe(16)
  expect(childContext.target.buffer.byteLength).toBe(32)
```

The only new assertion is this one:
```ts
 expect(grandchildContext.target.currentDate.toISOString()).toBe("2026-01-01T00:00:00.000Z")
```
But this one is also already implicitely tested by previous test case too (not at 2-level depth, but the test case name don't suggest this is what is being tested anyway)

Reply:
Done. I removed the previous test and renamed the current test as a means to improve clarity. The goal here is to test that the internal slots, memory, and/or date info. are not recreated potentially breaking internal functionality or expectations, especially with Date where every time a new Date is created you end up with a complete different date string.